cmd/thanos/downsample.go

```diff
@@ -81,6 +91,8 @@ func runDownsample(
 			if err := downsampleBucket(ctx, logger, bkt, dataDir); err != nil {
 				return errors.Wrap(err, "downsampling failed")
 			}
+
+			defer closeFn()
```

Comment:
Instead of this defer, we can put close in closeup section of run group -> below in `func(error){` body. This way we will close bucket on each error as well